sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -414,6 +399,22 @@ class Analyzer(
     }
 
     /**
+     * Build a project list for Project/Aggregate and expand the star if possible
+     */
+    private def buildExpandedProjectList(
+        exprs: Seq[NamedExpression],
+        plan: UnaryNode): Seq[NamedExpression] = {
+      exprs.flatMap {
+        // Using Dataframe/Dataset API: testData2.groupBy($"a", $"b").agg($"*")
+        case s: Star => s.expand(plan.child, resolver)
+        // Using SQL API without running ResolveAlias: SELECT * FROM testData2 group by a, b
+        case UnresolvedAlias(s: Star, _) => expandStarExpression(s, plan.child) :: Nil
```

Comment:
I think it should be `s.expand(plan.child, resolver)`, what you do here is just get rid of the outer `UnresolvedAlias` and resolve the star in next round.